### PR TITLE
Add a proof that (p + n) <= (p + m) => n <= m

### PR DESCRIPTION
**Pull request by justjoheinz**
Changed documentation to use <= instead of < when refering to `LTE`
Added total annotation to proofs

### CODE REVIEW

**Comment by justjoheinz** [comment]
I couldn' find anything in base or contrib which would have been a better place for contributions.
In general I think it would be a good idea to have something like contrib/Proofs/...

**Comment by Melvar** [comment]
That file already has `%default total`, no need to add more `total`s.

**Comment by ahmadsalim** [comment]
@justjoheinz Your code seems to fail type checking probably because you used `+` instead of `plus` for `lteAddBoth`. Could you kindly fix the error?

**Comment by justjoheinz** [comment]
@ahmadsalim shouldn't we setup something like `contrib/Proofs` in the first place. I feel a bit uneasy to commit to the Prelude.

**Comment by xekoukou** [comment]
I had the same unease with my simple proof. I have many more to come.
The difficulty though is not the proofs themselves but the categorization and the usefulness of the proofs.

https://github.com/idris-lang/Idris-dev/pull/3407#issuecomment-247976891

**Comment by ahmadsalim** [comment]
@justjoheinz Ah, thanks for the heads-up. I only checked the failing test and didn't investigate the rest of the code. Please do move the functions to `contrib` as suggested.

File: libs/prelude/Prelude/Nat.idr [hunk 1]
```diff
@@ -136,51 +136,57 @@ lteRefl : LTE n n
 lteRefl {n = Z}   = LTEZero
 lteRefl {n = S k} = LTESucc lteRefl
 
-||| n < m implies n < m + 1
+||| n <= m implies n < m + 1
```
Comment (SuprDewd) [inline libs/prelude/Prelude/Nat.idr]: "n < m + 1" should be "n <= m + 1"

File: libs/prelude/Prelude/Nat.idr [hunk 2]
```diff
@@ -136,51 +136,57 @@ lteRefl : LTE n n
...
 lteSuccRight : LTE n m -> LTE n (S m)
 lteSuccRight LTEZero     = LTEZero
 lteSuccRight (LTESucc x) = LTESucc (lteSuccRight x)
 
-||| n + 1 < m implies n < m 
+||| n + 1 <= m implies n < m
```
Comment (SuprDewd) [inline libs/prelude/Prelude/Nat.idr]: "n < m" should be "n <= m"

**Comment by ahmadsalim** [comment]
@justjoheinz Any updates?

**Comment by justjoheinz** [comment]
I will do that soon, had been caught up with other things.

**Comment by ahmadsalim** [comment]
I will close the PR since it seems that there has not been updates for a long time. If you are interested in finishing the PR, please reopen it with the code updates.

Thanks.

**Pull request closed without merging**